### PR TITLE
feat: us0008

### LINKED ISSUE #14: US - 0008
Criação da rota para `follower list` e `followed list` e implementação do serviço.

(ordem alfabetica - direta e inversa)

### PR DESCRIPTION
**Pull request by Victor-Planas**
feat: us0008 - created service to order seller follower list 
feat: added comparable to userDTO

### CODE REVIEW

File: src/main/java/br/com/mercadolivre/bootcampw2/grupo11/socialmeli/controllers/UsersController.java [hunk 1]
```diff
@@ -82,8 +82,9 @@ public GenericMessageDTO unfollow(@PathVariable @Min(0) Integer userId, @PathVar
 
     @GetMapping("/{userId}/followers/list")
     @ResponseStatus(HttpStatus.OK)
-    @Operation(description = "List all customers that follow a given seller")
-    public SellerFollowerListDTO listSellerFollowers(@PathVariable @Min(0) Integer userId){
-        return usersService.getFollowerList(userId);
+    @Operation(description = "List all customers that follow a given seller and a given order")
+    public SellerFollowerListDTO listSellerFollowers(@PathVariable @Min(0) Integer userId, @RequestParam(value = "order", defaultValue = "name_asc") String order) {
```
Comment (Giovannapls) [inline src/main/java/br/com/mercadolivre/bootcampw2/grupo11/socialmeli/controllers/UsersController.java]: Talvez seja interessante usar um Enum

File: src/main/java/br/com/mercadolivre/bootcampw2/grupo11/socialmeli/services/UsersService.java [hunk 2]
```diff
@@ -114,18 +118,28 @@ public void unfollow(Integer customerId, Integer sellerId) {
...
     /**
-     * This method is used to list all followers from a given seller
+     * This method is used to list all followers from a given seller with an optional given order
+     *
      * @param sellerId - seller ID
+     * @param order - order in which the follower list is ordered
      * @return Object DTO to the end user
      */
-    public SellerFollowerListDTO getFollowerList(Integer sellerId) {
+    public SellerFollowerListDTO getFollowerList(Integer sellerId, String order) {
```
Comment (Giovannapls) [inline src/main/java/br/com/mercadolivre/bootcampw2/grupo11/socialmeli/services/UsersService.java]: Mesmo de cima, talvez usar um Enum

File: src/main/java/br/com/mercadolivre/bootcampw2/grupo11/socialmeli/dtos/UserDTO.java [hunk 3]
```diff
@@ -15,4 +17,8 @@ public static UserDTO fromEntity(User user) {
         return new UserDTO(user.getUserId(), user.getUserName());
     }
 
+    @Override
+    public int compareTo(UserDTO o) {
+        return this.userName.toLowerCase().compareTo(o.getUserName().toLowerCase());
```
Comment (lmarizmeli) [inline src/main/java/br/com/mercadolivre/bootcampw2/grupo11/socialmeli/dtos/UserDTO.java]: check compareToIgnoreCase

**Review by lmarizmeli — APPROVED** [review]

**Review by Giovannapls — APPROVED** [review]
grand bravo :point_up:

**Review by CarlosFeldmann — APPROVED** [review]

**Review by CarlosFeldmann — APPROVED** [review]
🦧 monkey aproves 🦧

**Review by lmarizmeli — APPROVED** [review]
🐒

**Pull request merged**